### DISCUSSION TITLE
Compressing contents - best practice

### DISCUSSION THREAD

**Discussion by trixon** [AppImage/AppImageKit#1345] [General]
When creating appimages for my java applications I usually create a custom runtime image with `jlink`
jlink has an option to compress  the generated runtime, or not. See below.

From the perspective of the appimage performance, and perhaps the file size, what would be best to use?

And how would the be combined with the appimagetool --comp in the best way?

```
      --compress <compress>             Compression to use in compressing resources:
                                        Accepted values are:
                                        zip-[0-9], where zip-0 provides no compression,
                                        and zip-9 provides the best compression.
                                        Default is zip-6.

```

**Comment by probonopd** [comment]
Hello @trixon. A great question!

We want to conduct some systematic tests to find the optimal balance of parameters, please see

* https://github.com/AppImage/AppImageSpec/issues/44

I would assume that disabling jlink compression might be best, as the AppImage will handle compression on that level, and compressing the same data twice will probably only lead to slowdowns with little to no gains. It might also make zsync delta updates less efficient.

**Reply by trixon** [reply]
Thanks for the link @probonopd, I will follow that with interest.

I did some personal testing with different compressions with jlink and appimage about 4 years ago and if I remember correctly, it was the same then as the following results made today.

AppImage size with compressed (level 6) java runtime: 67MiB
AppImage size with no compressed (level 0) java runtime: 51MiB

Compressed custom java runtime image on disk: 76MiB
No compressed custom java runtime image  on disk: 130MiB

I did this without any options passed to appimagetool.

**Reply by TheAssassin** [reply]
You mean the AppImage, not the runtime, I presume. The runtime is just the binary header.

**Reply by trixon** [reply]
> You mean the AppImage, not the runtime, I presume. The runtime is just the binary header.

No, sorry about the confusion, I was talking about the java runtime.
Made the change to reflect that.
